include/envoy/http/filter.h

```diff
@@ -151,6 +151,14 @@ class StreamFilterCallbacks {
    */
   virtual Router::RouteConstSharedPtr route() PURE;
 
+  /**
+   * Schedules a request for a RouteConfiguration update from the management server.
+   * @param cb callback to be called when the configuration update has been propagated to the worker
+   * thread.
+   * @return whether RouteConfiguration update has been scheduled.
+   */
+  virtual bool requestRouteConfigUpdate(std::function<void()> cb) PURE;
```

Comment:
Here and elsewhere, could you give these callbacks more descriptive names? Maybe this one could be route_config_updated_cb or something.

Reply:
done.